opentech/apply/funds/models.py

```diff
@@ -513,7 +493,36 @@ def current(self):
         return self.exclude(next__isnull=False)
 
 
-class ApplicationSubmission(WorkflowHelpers, BaseStreamForm, AbstractFormSubmission):
+class AddTransitions(models.base.ModelBase):
+    def __new__(cls, name, bases, attrs, **kwargs):
+        transition_prefix = 'transition'
+        for workflow in WORKFLOWS.values():
+            for phase, data in workflow.items():
+                for transition_name, action in data.all_transitions.items():
+                    method = data.transition_methods.get(transition_name)
+                    # Get the method defined on the parent or default to a NOOP
+                    transition_state = attrs.get(method, lambda self: None)
+                    # Provide a neat name for graph viz display
+                    function_name = '_'.join([transition_prefix, slugify(action)])
+                    transition_state.__name__ = function_name
+                    # Wrap with transition decorator
+                    transition_func = transition(attrs['status'], source=phase, target=transition_name)(transition_state)
+
+                    # Attach to new class
+                    method_name = '_'.join([transition_prefix, transition_name, str(data.step)])
+                    attrs[method_name] = transition_func
+
+        def get_transition(self, transition):
+            return getattr(self, '_'.join([transition_prefix, transition, str(self.phase.step)]))
+
+        attrs['get_transition'] = get_transition
+
+        # attrs['restart'] = transition(attrs['status'], source='*', target=INITIAL_STATE)(lambda x: None)
```

Comment:
remove now that we have cb66215 ?

Reply:
cb66215 doesnt work with the old style statuses . I had written this when i realised it wouldn't work.

Reply:
👍